pkg/registry/namespace/etcd/etcd.go

```diff
@@ -99,21 +101,67 @@ func (r *REST) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 
 	namespace := nsObj.(*api.Namespace)
 
+	// Ensure we have a UID precondition
```

Comment:
UID preconditions didn't exist in 1.2, I'll adjust appropriately when I cherry pick